### PR TITLE
feat(storefrontlib, core): foundation of lazy loaded translations 

### PR DESCRIPTION
**Pull request by Platonn**
fixes GH-1629

### CODE REVIEW

File: projects/core/src/i18n/translate.pipe.spec.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  let pipe: TranslatePipe;
+  let service: TranslationService;
+  let cd: ChangeDetectorRef;
+  let mockServiceLanguageChanged$: Subject<string>;
+
+  beforeEach(() => {
+    mockServiceLanguageChanged$ = new Subject();
+    const mockTranslateService = {
+      translateLazy: () => {},
+      languageChanged$: mockServiceLanguageChanged$
```
Comment (dunqan) [inline projects/core/src/i18n/translate.pipe.spec.ts]: is not needed and not present on Translation service anymore

File: projects/core/src/i18n/translate.pipe.spec.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+describe('TranslatePipe', () => {
+  let pipe: TranslatePipe;
+  let service: TranslationService;
+  let cd: ChangeDetectorRef;
+  let mockServiceLanguageChanged$: Subject<string>;
+
+  beforeEach(() => {
+    mockServiceLanguageChanged$ = new Subject();
+    const mockTranslateService = {
```
Comment (dunqan) [inline projects/core/src/i18n/translate.pipe.spec.ts]: Maybe `mockTranslationService` to make it consistent with the subject of mocking

File: projects/core/src/i18n/translation.service.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  constructor(
+    private i18NextService: I18NextService,
+    private config: ServerConfig
+  ) {}
+
+  exists(key: string, options: any = {}): boolean {
+    return this.i18NextService.exists(key, options);
+  }
+
+  translate(key: string, options: any = {}): string {
```
Comment (dunqan) [inline projects/core/src/i18n/translation.service.ts]: Should we use this one by default, or rather `translateLazy`?

File: projects/core/src/i18n/translation.service.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  }
+
+  translateLazy(key: string, options: any = {}): Observable<string> {
+    return new Observable<string>(subscriber => {
+      if (this.i18NextService.exists(key, options)) {
+        subscriber.next(this.i18NextService.t(key, options));
+        subscriber.complete();
+      } else {
+        this.loadKeyNamespace(key, () => {
```
Comment (dunqan) [inline projects/core/src/i18n/translation.service.ts]: Just a question: do we want to display fallback (nbsp) while translations are loading?

File: projects/core/src/i18n/translation.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+      beforeEach(() => {
+        i18NextService.exists.and.returnValue(true);
+      });
+
+      it('should emit result of i18NextService.t', () => {
+        i18NextService.t.and.returnValue('value');
+        let result;
+        service
+          .translateLazy(testKey, testOptions)
+          .pipe(take(1))
```
Comment (dunqan) [inline projects/core/src/i18n/translation.service.spec.ts]: pipe(take(1)) is probably not needed here anymore (and in further usages)

File: projects/core/src/i18n/translation.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+const testKey = 'testNamespace:testKey';
+const testOptions = 'testOptions';
+const nonBreakingSpace = String.fromCharCode(160);
+
+describe('TranslationService', () => {
+  let service: TranslationService;
+  let config: ServerConfig;
+  let i18NextService;
+  let mockI18NextLanguage;
```
Comment (dunqan) [inline projects/core/src/i18n/translation.service.spec.ts]: It is not used anymore I believe

File: projects/core/src/i18n/translation.service.ts (hunk 4 above)
Comment (Platonn) [reply projects/core/src/i18n/translation.service.ts]: good point

File: projects/core/src/i18n/translation.service.ts (hunk 3 above)
Comment (Platonn) [reply projects/core/src/i18n/translation.service.ts]: Removed synchronous translate as it's unpredictable. And renamed translateLazy to translate.

**Pull request merged**